scripts/irods/test/test_izonereport.py

```diff
@@ -86,14 +92,108 @@ def test_izonereport_and_validate(self):
+            self.assertGreaterEqual(len(zone_info['coordinating_resources']), 1)
+            for resource in zone_info['coordinating_resources']:
                 self.assertIn('id', resource.keys())
+
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing or python rule engine")
+    def test_servers_are_flattened_core__issue_6857(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        self.assertIn('servers', zone_info.keys())
+        self.assertNotIn('catalog_server', zone_info.keys())
+        self.assertEqual(len(zone_info['servers']), 1)
+        
+        self.assertIn('coordinating_resources', zone_info.keys())
+        self.assertEqual(len(zone_info['coordinating_resources']), 3)
+        
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python' or not test.settings.RUN_IN_TOPOLOGY, "Skip for Core Testing or python rule engine")
+    def test_servers_are_flattened_topology__issue_6857(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        self.assertIn('servers', zone_info.keys())
+        self.assertNotIn('catalog_server', zone_info.keys())
+        self.assertEqual(len(zone_info['servers']), 4)
+        
+        self.assertIn('coordinating_resources', zone_info.keys())
+        self.assertEqual(len(zone_info['coordinating_resources']), 3)
+        
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python', "only run for native rule language")
+    def test_zone_report_calculates_checksums_for_all_plugins__issue_6856(self):
+        
+        _, stdout, _ = self.admin.assert_icommand('izonereport', 'STDOUT')
+        
+        zone_info = json.loads(stdout)['zones'][0]
+        
+        for server in zone_info['servers']:
+            for plugin in server['plugins']:
+                
+                # Skip database plugin which only exists on catalog server when running on consumer server
+                if plugin['type'] == 'database' and test.settings.TOPOLOGY_FROM_RESOURCE_SERVER:
+                    continue
+                
+                plugin_name = f'lib{plugin["name"]}.so'
+                
+                # Find folder containing plugin file
+                plugin_folder = ""
+                for root, _dirs, files in os.walk(paths.plugins_directory()):
+                    if plugin_name in files:
+                        plugin_folder = root
+                        break
+                    
+                plugin_path = os.path.join(plugin_folder, plugin_name)
+                plugin_checksum = lib.file_digest(plugin_path, 'sha256', encoding='base64')
+                
+                self.assertEqual(plugin['checksum_sha256'][5:], plugin_checksum)
```

Comment:
not sure why the `[5:]` here... please comment or replace with a named variable.

Reply:
I agree, it is unclear. the '[5:]' is used to remove the 'sha2:' prefix that is added when a checksum is calculated using the irods checksum functions.

I modified the code to be clearer:
```python
# Check that zone report checksum without 'sha2:' prefix is equal to the expected
self.assertEqual(plugin['checksum_sha256'][len('sha2:'):], expected_plugin_checksum)
```

The other alternative would be to remove the prefix from the checksums in the zone report.

Reply:
if we change the algorithm, we'd change the name of the key/property.  so yes, i think we can remove the 'sha2:' in the value in the zonereport json.